Source/Invocation.cs

```diff
@@ -70,14 +74,19 @@ protected Invocation(MethodInfo method, params object[] arguments)
 		/// </summary>
 		public MethodInfo Method => this.method;
 
+		/// <summary>
+		/// Gets the arguments of the invocation.
+		/// </summary>
+		public IReadOnlyList<object> Arguments => this.ArgumentsArray;
```

Comment:
Exposing the underlying array as-is doesn't make it magically read only heart because the interface says so. But I guess it's not critical anyway.

Reply:
I think this is a failure of the interface's name; it's a readable list, rather than one which can only be read. If it needs to be immutable, a copy-on-read method could be used.

Reply:
I don't see the practical benefits of making recorded invocations truly immutable; if someone feels the need to cast an interface back to the underlying runtime type, why not just let them? The fact that we're using an interface should be a strong enough signal against doing that. IMHO casting to an interface is good enough in practice.